src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.ClientCertificates.cs

```diff
@@ -104,69 +97,81 @@ public async Task Manual_SSLBackendNotSupported_ThrowsPlatformNotSupportedExcept
-            }
-            else if (certIndex == 2)
-            {
-                // This is a valid client cert since it has no EKU thus all usages are permitted.
-                clientCert = Configuration.Certificates.GetNoEKUCertificate();
-            }
-            else if (certIndex == 3)
-            {
-                // This is an invalid client cert since it has an EKU but is missing ClientAuthentication OID.
-                clientCert = Configuration.Certificates.GetServerCertificate();
-            }
-
-            Assert.NotNull(clientCert);
+            var options = new LoopbackServer.Options { UseSsl = true };
 
-            HttpClientHandler handler = CreateHttpClientHandler(useSocketsHttpHandlerString, useHttp2String);
-            handler.ClientCertificates.Add(clientCert);
-            using (HttpClient client = CreateHttpClient(handler, useHttp2String))
+            X509Certificate2 GetClientCertificate(int certIndex)
             {
-                var request = new HttpRequestMessage();
-                request.RequestUri = new Uri(Configuration.Http.EchoClientCertificateRemoteServer);
-
-                // Issue #35239. Force HTTP/1.1.
-                request.Version = new Version(1,1);
-                HttpResponseMessage response = client.SendAsync(request).GetAwaiter().GetResult(); // need a 4-arg overload of RemoteInvoke that returns a Task
-                Assert.Equal(expectedStatusCode, response.StatusCode);
+                X509Certificate2 cert = null;
 
-                if (expectedStatusCode == HttpStatusCode.OK)
+                // Get client certificate. RemoteInvoke doesn't allow easy marshaling of complex types.
+                // So we have to select the certificate at this point in the test execution.
+                if (certIndex == 1)
                 {
-                    string body = response.Content.ReadAsStringAsync().GetAwaiter().GetResult(); // need a 4-arg overload of RemoteInvoke that returns a Task
-                    byte[] bytes = Convert.FromBase64String(body);
-                    var receivedCert = new X509Certificate2(bytes);
-                    Assert.Equal(clientCert, receivedCert);
+                    // This is a valid client cert since it has an EKU with a ClientAuthentication OID.
+                    cert = Configuration.Certificates.GetClientCertificate();
                 }
+                else if (certIndex == 2)
+                {
+                    // This is a valid client cert since it has no EKU thus all usages are permitted.
+                    cert = Configuration.Certificates.GetNoEKUCertificate();
+                }
+                else if (certIndex == 3)
+                {
+                    // This is an invalid client cert since it has an EKU but is missing ClientAuthentication OID.
+                    cert = Configuration.Certificates.GetServerCertificate();
+                }
+
+                return cert;
```

Comment:
Nit: this can be simplified if desired, e.g.
```C#
X509Certificate2 GetClientCertificate(int certIndex) => certIndex switch
{
    1 => Configuration.Certificates.GetClientCertificate(),
    2 => Configuration.Certificates.GetNoEKUCertificate(),
    3 => Configuration.Certificates.GetServerCertificate(),
    _ => null
}
```
(plus comments of course)

Reply:
I'll switch to that. Thanks for teaching me new C# syntax. :)